port/unix/omrsysinfo.c

```diff
@@ -4215,6 +4248,71 @@ omrsysinfo_cgroup_is_memlimit_set(struct OMRPortLibrary *portLibrary)
 #endif /* defined(LINUX) && !defined(OMRZTPF) */
 }
 
+intptr_t
+omrsysinfo_cgroup_get_handle_subsystem_file(struct OMRPortLibrary *portLibrary,  uint64_t subsystemFlag, const char *fileName)
+{
+	intptr_t fd = 0;
+#if defined(LINUX) && !defined(OMRZTPF)
+	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
+	char fullPathBuf[PATH_MAX];
+	char *fullPath = fullPathBuf;
+	intptr_t bufferLength = PATH_MAX;
+	BOOLEAN allocateMemory = FALSE;
+	rc = getAbsolutePathOfCgroupSubsystemFile(portLibrary, subsystemFlag, fileName, fullPath, &bufferLength, &allocateMemory);
+	if (allocateMemory) {
+		fullPath = portLibrary->mem_allocate_memory(portLibrary, bufferLength, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL == fullPath) {
+			Trc_PRT_readCgroupSubsystemFile_oom_for_filename();
+			rc = portLibrary->error_set_last_error_with_message(portLibrary, OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED, "memory allocation for filename buffer failed");
+			goto _end;
+
+		}
+		rc = getAbsolutePathOfCgroupSubsystemFile(portLibrary, subsystemFlag, fileName, fullPath, &bufferLength, &allocateMemory);
+	}
+	if (rc == 0) {
+		fd = portLibrary->file_open(portLibrary, fullPath, EsOpenRead, 0);
+	}
+	if (allocateMemory) {
+		portLibrary->mem_free_memory(portLibrary, fullPath);
+	}
+	
+#endif /* defined(LINUX) && !defined(OMRZTPF) */
+	goto _end;
+_end:
```

Comment:
If you move this label inside the `#if` block, the `goto` immediately preceding it can be removed.

Reply:
i have removed `goto`. Thanks